db/seeds.d/111-upgrade_tasks.rb

```diff
@@ -0,0 +1,9 @@
+UpgradeTask.define_tasks(:katello) do
+  [
+    {:name => 'katello:correct_repositories', :long_running => true, :skip_failure => true, :always_run => true},
+    {:name => 'katello:correct_puppet_environments', :long_running => true, :skip_failure => true, :always_run => true},
+    {:name => 'katello:clean_backend_objects', :long_running => true, :skip_failure => true, :always_run => true},
+    {:name => 'katello:clean_backend_objects', :long_running => true, :skip_failure => true, :always_run => true},
+    {:name => 'katello:upgrades:3.7:make_all_ks_repos_bootable', :long_running => true}
```

Comment:
for rake upgrades in the past, it looks like we were running all the upgrade tasks, even from 3.0. Is that true, and are they all designed to be idempotent? If so, why not just include all those tasks in the new plugin upgrades? Why just include the 3.7 one?

Reply:
They are designed to be idempotent, but most upgrade tasks are only run once.  The fact that those once-only tasks means that we were just keeping them around for no reason, in addition to that:

1.  they would get run on first upgrade, even if not needed (a katello 2.3 upgrade task would get run when installing a fresh katello 3.7 and then running --upgrade the first time).
2.  We don't support upgrading from such an older version to such a newer version.  In the past we've guided, we support upgrades from N-2, but we haven't even tested that, generally we only test N-1. 

Maybe we should add the deletion of old upgrade tasks to the release steps?

Reply:
Hmm...  I'm ok with adding to the release process, if it's easy. What about a rake task to remove old upgrade tasks based on the VERSION?

Reply:
It would have to somehow edit this seeds file, which seems kinda weird.

Reply:
and actually, i meant to remove the 3.7 upgrade task since this won't make it till 3.9, let me do that.

Reply:
If we add a trailing comma, it'd be easier to write automation in tool_belt to do this.

Reply:
added, and accidentally had clean_backend_objects in there twice, so i removed that :)